modules/sdk-coin-doge/src/doge.ts

```diff
@@ -10,13 +10,51 @@ import {
   CrossChainRecoverySigned,
   CrossChainRecoveryUnsigned,
   RecoverFromWrongChainOptions,
+  TransactionInfo,
+  TransactionPrebuild,
 } from '@bitgo/abstract-utxo';
 import { BaseCoin, BitGoBase, HalfSignedUtxoTransaction, SignedTransaction } from '@bitgo/sdk-core';
-import * as utxolib from '@bitgo/utxo-lib';
+import { bitgo, networks } from '@bitgo/utxo-lib';
+
+type UnspentJSON = bitgo.Unspent<number> & { valueString: string };
+type TransactionInfoJSON = TransactionInfo<number> & { unspents: UnspentJSON[] };
+type TransactionPrebuildJSON = TransactionPrebuild<number> & { txInfo: TransactionInfoJSON };
+
+function parseUnspents<TNumber extends number | bigint>(
+  unspents: UnspentJSON[] | bitgo.Unspent<TNumber>[]
+): bitgo.Unspent<bigint>[] {
+  return unspents.map((unspent: bitgo.Unspent<TNumber> | UnspentJSON): bitgo.Unspent<bigint> => {
+    if (typeof unspent.value === 'bigint') {
+      return unspent as bitgo.Unspent<bigint>;
+    }
+    if ('valueString' in unspent) {
+      return { ...unspent, value: BigInt(unspent.valueString) };
+    }
+    if (typeof unspent.value === 'number') {
+      throw new Error(`received Unspent<number> where Unspent<bigint> or UnspentJSON was expected`);
+    }
+    throw new Error('invalid unspent');
```

Comment:
This could be a bit more specific to say that the unspent value was not in the right format.

As an aside, why does this method signature accept `TNumber` if it throws on `number` values, shouldn't it just be `Unspent<bigint>` in the parameter signature?

Reply:
strictly speaking yes, but TNumber was required to make typescript happy here

Reply:
actually we can use bigint instead of TNumber in parse functions atleast. But then we need to pass arguments as `params.txPrebuild as TransactionPrebuildJSON`.  Please check commit https://github.com/BitGo/BitGoJS/pull/2908/commits/53854dc76bd03fe958ac2d6e11eb121d1ce44c50 if it looks better or else I can revert to older commit

Reply:
> But then we need to pass arguments as params.txPrebuild as TransactionPrebuildJSON

that would not a legitimate cast though because we cannot expect `txPrebuild` to always be of that type

Reply:
alright, reverting to the older commit